pyuvdata/uvdata/uvdata.py

```diff
@@ -5335,71 +5334,60 @@ def _apply_w_proj(self, new_w_vals, old_w_vals, select_mask=None):
             return
 
         if select_mask is None:
-            select_len = self.Nblts
+            select_mask = ...
         else:
             try:
                 inv_mask = np.ones(self.Nblts, dtype=bool)
                 inv_mask[select_mask] = False
-                select_mask = ~inv_mask
-                select_len = np.sum(select_mask)
+
+                if all(inv_mask):
+                    # If nothing is selected, then bail
+                    return
+
+                # If everything is selected by the mask, then no entries pop up in the
+                # inverse, in which case it's faster to use the Ellipsis
```

Comment:
very helpful comment

Reply:
Definitely a happy accidental discovery from the past few months..`Ellipsis`